src/Features/CSharp/Portable/RemoveAsyncModifier/CSharpRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            // For async void methods this fixer does the same as Make Method Synchronous so we don't need to offer both
+            if (methodSymbol.ReturnsVoid)
+            {
+                return false;
+            }
+
+            // IAsyncEnumerable iterators cannot be made non-async even if they don't have awaits
+            return !methodSymbol.ReturnType.OriginalDefinition.Equals(knownTypes._iAsyncEnumerableOfTTypeOpt);
+        }
+
+        protected override SyntaxNode GetLastChildOfBlock(SyntaxNode node)
+            => GetBlockBody(node).ChildNodes().Last();
+
+        protected override ControlFlowAnalysis AnalyzeControlFlow(SemanticModel semanticModel, SyntaxNode node)
+            => semanticModel.AnalyzeControlFlow(GetBlockBody(node));
+
+        private static SyntaxNode GetBlockBody(SyntaxNode node)
+            => node switch
+            {
+                MethodDeclarationSyntax method => method.Body,
+                LocalFunctionStatementSyntax localFunction => localFunction.Body,
+                AnonymousFunctionExpressionSyntax anonymousFunction => anonymousFunction.Block,
+                _ => throw ExceptionUtilities.Unreachable
+            };
+
+        protected override bool TryGetExpressionBody(SyntaxNode node, out SyntaxNode expression)
+        {
+            expression = node switch
+            {
+                // For methods and local functions ExpressionBody is an ArrowExpressionClauseSyntax so we pull the real expression out
+                MethodDeclarationSyntax method => method.ExpressionBody?.Expression,
+                LocalFunctionStatementSyntax localFunction => localFunction.ExpressionBody?.Expression,
+                AnonymousFunctionExpressionSyntax anonymousFunction => anonymousFunction.ExpressionBody,
+                _ => throw ExceptionUtilities.Unreachable
+            };
+
+            return expression != null;
+        }
+
+        protected override SyntaxNode ConvertToBlockBody(SyntaxNode node, SyntaxNode expressionBody, SyntaxEditor editor)
+        {
+            if (InitializeParameterHelpers.TryConvertExpressionBodyToStatement(expressionBody, SyntaxFactory.Token(SyntaxKind.SemicolonToken), false, out var statement))
+            {
+                var block = SyntaxFactory.Block(statement);
+                return node switch
+                {
+                    MethodDeclarationSyntax method => method.WithBody(block).WithExpressionBody(null).WithSemicolonToken(default),
+                    LocalFunctionStatementSyntax localFunction => localFunction.WithBody(block).WithExpressionBody(null).WithSemicolonToken(default),
+                    AnonymousFunctionExpressionSyntax anonymousFunction => anonymousFunction.WithBody(block).WithExpressionBody(null),
+                    _ => throw ExceptionUtilities.Unreachable
+                };
+            }
+            return null;
```

Comment:
```suggestion
            }
            
            return null;
```

Nit: after a statement that ends with a `}` we prefer a blank line.  It's on me to actually make an analyzer/fixer to catch this, i just haven't had time.